src/database-layer/db.ts

```diff
@@ -127,14 +130,31 @@ if (process.env.TRANSACTION_TIMEOUT_MS) {
 } else {
 	timeoutMS = 10000
 }
+
+const rejectedExecuteSql = memoize(
+	(message: string): Tx['executeSql'] =>
+		(_sql, _bindings, callback) =>
+			Promise.reject(new Error(message)).nodeify(callback),
```

Comment:
If I get it right, the principle is to only create `new Error()`s once per error message.
But, the memoized function returns high order function that will call `new Error()` in each call.
Should we use a closure for the rejected error?
Am I missing something?

Reply:
Using a closure for it would make sense actually and I'll make that change - in reality though the difference should be minimal as it should be a very rare error case to actually call one of these functions after the transaction has been closed

Reply:
Actually thinking about it a bit more I think it makes sense as-is, it would be better performance to closure-scope the error but only in error cases which shouldn't be happening in production mode anyway, and closure-scoping it would make the stack trace completely useless (whereas it would still be useful if it's created on-demand)

Reply:
So the whole point of memoize that reduces the overhead is to create those two functions less often? I thought that the `new Error()` was the costly operation.

Reply:
Yeah, `new Error()` is pretty costly (although `err.stack` is even more costly the first time it's called).  The memoisation was just to avoid constantly recreating lambdas but it really shouldn't matter in this case - I'll benchmark it tomorrow and probably end up removing it

Reply:
Removing it doesn't seem to matter so I've done that